script.tag-generator/main.py

```diff
@@ -1,120 +1,141 @@
-import re, urllib2, urllib, httplib, sys, os, time
-import xbmc 
+import datetime
+import json
+import os
+import re
+import sys
+import xbmc
 import xbmcgui 
 import xbmcaddon
-import json
 try:
+    import requests
     import simplejson
     import trakt
     from trakt import users
+    from bs4 import BeautifulSoup
 except:
     if os.name == "nt":
         slash = "\\"
     else:
         slash = "/"
     sys.path.append(os.path.abspath(os.path.dirname(__file__)+slash+"resources"+slash+"lib"))
+    import requests
     import simplejson
     import trakt
     from trakt import users
+    from bs4 import BeautifulSoup
 
 try:
     from hashlib import sha1
 except ImportError:
     from sha import new as sha1
 
+
 __settings__ = xbmcaddon.Addon()
 __language__ = __settings__.getLocalizedString
 c_refresh = __settings__.getSetting("32012")
-c_runasservice = __settings__.getSetting("32011")
-sleeptime = int(c_refresh)*3600000
-micronap = 60000
+c_runasservice = json.loads(__settings__.getSetting("32011"))
+start_time = datetime.datetime.now()
+service_interval = int(c_refresh)*360  # seconds
+micronap = 500  # milliseconds
+
 
-#I want my strings back please
 def _getstr(id):
     return str(__language__(id))
 
 if len(sys.argv) != 2:
     xbmc.sleep(1000)
-    xbmc.log(msg=_getstr(30002),level=xbmc.LOGNOTICE)
+    xbmc.log(msg="TAG-GEN: Starting as a service.", level=xbmc.LOGNOTICE)
 
 ###################################################################
 ############################ FUNCTIONS ############################
 ###################################################################
-#test for interwebs
+# test for interwebs
 def internet_test(url):
     try:
-        response=urllib2.urlopen(url,timeout=1)
+        response = requests.get(url, timeout=5)
         return True
-    except urllib2.URLError as err: pass
+    except Exception as e:
+        pass
     if len(sys.argv) == 2:
         dialog = xbmcgui.Dialog() 
         ok = dialog.ok(_getstr(30000),url + _getstr(30001))
-    xbmc.log(msg= "TAG-GEN: " + url + _getstr(30001),level=xbmc.LOGNOTICE)
-    sys.exit(url + _getstr(30001))
+    xbmc.log(msg="TAG-GEN: " + str(url) + " unreachable. Check network and retry.", level=xbmc.LOGERROR)
+    sys.exit(1)
+
 
 # cancels script
 def ifcancel():
     if len(sys.argv) == 2:
         if (pDialog.iscanceled()):
-            xbmc.log(msg= _getstr(30003),level=xbmc.LOGNOTICE)
-            sys.exit(_getstr(30003))
+            xbmc.log(msg="TAG-GEN: Cancel received from Kodi dialog, exiting.", level=xbmc.LOGNOTICE)
+            sys.exit(0)
+
 
 #def to make a debug log
 def debuglog(string):
-    if "true" in c_debug:
-        xbmc.log(msg=string,level=xbmc.LOGNOTICE)
+    if c_debug:
+        xbmc.log(msg=string, level=xbmc.LOGNOTICE)
```

Comment:
If I do that then a user has to go into System and enable debug mode as well as enabling it in my addon. This would also make it much harder to find something from tag-gen.

Reply:
You should request your users to provide a debug log if you need to debug your code